frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx

```diff
@@ -247,21 +247,27 @@ export const usePipelineTaskTopology = (
   executions?: Execution[] | null,
   events?: Event[] | null,
   artifacts?: Artifact[],
-): PipelineNodeModelExpanded[] =>
+): PipelineNodeModelExpanded[] | Error =>
```

Comment:
why are you returning an error here?

You should instead follow our previous ways of handling state and return [data, loaded, error] if we need that